EssentialFeed/EssentialFeedTests/Feed API/LoadFeedImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,243 @@
+//
+//  LoadFeedImageCommentsFromRemoteUseCaseTests.swift
+//  EssentialFeedTests
+//
+//  Created by Ivan Ornes on 9/3/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class LoadFeedImageCommentsFromRemoteUseCaseTests: XCTestCase {
+	
+	func test_init_doesNotRequestDataFromURL() {
+		let (_, client) = makeSUT()
+		
+		XCTAssertTrue(client.requestedURLs.isEmpty)
+	}
+	
+	func test_loadImageCommentsFromURL_requestsCommentsFromURL() {
+		let imageID = UUID().uuidString
+		let url = URL(string: "https://a-given-url.com/image/\(imageID)/comments")!
+		let (sut, client) = makeSUT()
+		
+		_ = sut.loadImageComments(imageID: imageID) { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url])
+	}
+	
+	func test_loadImageCommentsFromURLTwice_requestsCommentsFromURLTwice() {
+		let imageID = UUID().uuidString
+		let url = URL(string: "https://a-given-url.com/image/\(imageID)/comments")!
+		let (sut, client) = makeSUT()
+		
+		_ = sut.loadImageComments(imageID: imageID) { _ in }
+		_ = sut.loadImageComments(imageID: imageID) { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_loadImageCommentsFromURL_deliversConnectivityErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		let clientError = NSError(domain: "a client error", code: 0)
+		
+		expect(sut, toCompleteWith: failure(.connectivity), when: {
+			client.complete(with: clientError)
+		})
+	}
+	
+	func test_loadImageCommentsFromURL_deliversInvalidDataErrorOnNon2xxHTTPResponse() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [199, 150, 300, 400, 500]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				client.complete(withStatusCode: code, data: anyData(), at: index)
```

Comment:
`anyData` is not a valid response for this endpoint based on API the contract. It should use a valid response to be able to test the `status code` logic.

It could be an empty JSON list: 

```suggestion
				client.complete(withStatusCode: code, data: makeItemsJSON([]), at: index)